### ISSUE TITLE
[Hunter] Aspect of the Viper/Дух гадюки

### ISSUE THREAD

**Issue by Ta-tsu-ka** [moonwelltbc/bugtracker#45]
### Описание бага

не работает Аспект https://db.moonwell.su/?spell=34074
фулл мана без аспекта
![WoWScrnShot_080723_111113](https://github.com/moonwelltbc/bugtracker/assets/138556170/e0043d76-ae59-4498-8072-63ad3e64ab16)
фулл мана с аспектом
![WoWScrnShot_080723_111119](https://github.com/moonwelltbc/bugtracker/assets/138556170/307decd2-d9fe-4bf1-abd0-44b15fdf2d2e)
без маны без аспекта
![WoWScrnShot_080723_111401](https://github.com/moonwelltbc/bugtracker/assets/138556170/6c16a3ee-02b0-4eae-accf-49139ed06d0e)
без маны с аспектом
![WoWScrnShot_080723_111407](https://github.com/moonwelltbc/bugtracker/assets/138556170/7864153f-b14a-44d0-84b6-ce8a7894c52a)

так же не работает бонус от 2 т6 или просто баг в описании "Охотник обращается к духу гадюки, восполняя ману в объёме до 55%* своего интеллекта..."

### Как воспроизвести баг?

1. включить аспект

### Как должно работать?

как работает на актуальном ядре
фулл мана без аспекта
![WoWScrnShot_080723_003920](https://github.com/moonwelltbc/bugtracker/assets/138556170/355ced7e-516c-4f01-8aea-c61ee7821a58)
фулл мана с аспектом
![WoWScrnShot_080723_003924](https://github.com/moonwelltbc/bugtracker/assets/138556170/9eba8634-ab78-4144-933e-8adebefac4bd)
без маны без аспекта
![WoWScrnShot_080723_004150](https://github.com/moonwelltbc/bugtracker/assets/138556170/893f4f29-da86-4ef4-b584-04f3439424f4)
без мана с аспектом
![WoWScrnShot_080723_004154](https://github.com/moonwelltbc/bugtracker/assets/138556170/8f40eb94-89cb-4c7e-a1e4-7b58eb91cea6)

описание с бонусом от 2 т6 "Охотник обращается к духу гадюки, восполняя ману в объёме до 60%* своего интеллекта..."

### Игровой мир

Moonwell x100

### Версия ядра

12fc663 (2023-08-06 16:07:28)

**Comment by Gensen1** [comment]
Все работает, восстанавливает дополнительно ману в правильном количестве и учитывает бонус т6.
![image](https://github.com/moonwelltbc/bugtracker/assets/64833670/ede27417-41fe-4963-8117-5dfaee2c13e8)

**Comment by Gensen1** [comment]
Да, действительно, формула не учитывала текущий объем маны. Необходимо было это уточнить в первом сообщении.

Теперь расчет будет происходить по формуле: 
`intellect * 22.0f / 35.0f * (0.9f - mana / max_mana) + intellect * 0.11f + 0.35f * level`

**Comment by Ta-tsu-ka** [comment]
> Да, действительно, формула не учитывала текущий объем маны. Необходимо было это уточнить в первом сообщении.

всё ещё не работает.
